bundle/config/bundle.go

```diff
@@ -43,4 +43,10 @@ type Bundle struct {
 
 	// Deployment section specifies deployment related configuration for bundle
 	Deployment Deployment `json:"deployment,omitempty"`
+
+	// Minimum Databricks CLI version required to run the bundle.
+	MinDatabricksCliVersion string `json:"min_databricks_cli_version,omitempty"`
+
+	// Maximum Databricks CLI version required to run the bundle.
+	MaxDatabricksCliVersion string `json:"max_databricks_cli_version,omitempty"`
```

Comment:
Note: Typically when specifying a max version as a user I would like to also permit patch releases under the same minor release. This is not allowed right now. 

We can permit this by having the max version be exclusionary. Meaning the requirements are:
1. currentVersion >= minVersion
2. currentVersion < maxVersion

This for example allowed me to set maxVersion to `2.0.0` and be sure that no breaking changes will take place with the CLI versions I do permit.

Reply:
@shreyas-goenka that's a good point on allowing patch releases. But I disagree with "strictly lower" proposal.
when reading the line `max_databricks_cli_version: 2.0.0` I'm reading this as "I can use 2.0.0 but not any version after". Also strictly lower approach does not allow you to define "this is the last CLI version with which bundle work". _(strictly speaking it allows but it will look unintuitive as you would need to provide next version instead)_

Alternative approach is to use 3rd party libraries like this https://github.com/Masterminds/semver to do the version constraint checks

Reply:
Instead of a hard min/max (with the inclusive/exclusive ambiguity), did you take a look at using version constraints often used in dependency management?

Then one could write `~= 0.216` to mean `>= 0.216 && == 0.*` (or `>= 0.216 && < 1`).

See for example https://packaging.python.org/en/latest/specifications/version-specifiers/#id5

Reply:
Another example: https://developer.hashicorp.com/terraform/language/expressions/version-constraints#version-constraint-syntax

Reply:
@pietern yes, the link I shared above was about version constraints checks https://github.com/Masterminds/semver?tab=readme-ov-file#checking-version-constraints

let's discuss this offline quickly to make sure we're all on the same page here

Reply:
Done in the following commit https://github.com/databricks/cli/pull/1320/commits/bcbf6489b58c0e9f992b88152a1ca02172e33cc5